### PR TITLE
[TACHYON-1284] make security socket timeout configurable.

### PR DESCRIPTION
**Pull request by csliu**
as the title.

### CODE REVIEW

**Comment by gpang** [comment]
I don't think your commit is associated with your github account. Could you follow the instructions here: https://help.github.com/articles/setting-your-email-in-git/ 

Thanks!

File: common/src/main/resources/tachyon-default.properties [hunk 1]
```diff
@@ -54,6 +54,7 @@ tachyon.integration.worker.resource.mem=1024MB
 
 # Security properties
 tachyon.security.authentication.type=NOSASL
+tachyon.security.authentication.socket.timeout_second=30
```
Comment (apc999) [inline common/src/main/resources/tachyon-default.properties]: how about rename this property to `tachyon.security.authentication.socket.timeout.second`?

File: common/src/main/java/tachyon/Constants.java [hunk 2]
```diff
@@ -336,6 +336,8 @@
   public static final String SECURITY_AUTHENTICATION_TYPE = "tachyon.security.authentication.type";
   public static final String SECURITY_AUTHENTICATION_CUSTOM_PROVIDER =
       "tachyon.security.authentication.custom.provider.class";
+  public static final String SECURITY_AUTHENTICATION_SOCKET_TIMEOUT =
+      "tachyon.security.authentication.socket.timeout_second";
```
Comment (apc999) [inline common/src/main/java/tachyon/Constants.java]: change this correspondingly.

File: minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java [hunk 3]
```diff
@@ -203,6 +203,8 @@ public TachyonConf newTestConf() throws IOException {
       testConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, level),
           Joiner.on(',').join(newPaths));
     }
+
+    testConf.set(Constants.SECURITY_AUTHENTICATION_SOCKET_TIMEOUT, "30");
```
Comment (apc999) [inline minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java]: unnecssary

**Comment by apc999** [comment]
Thanks @csliu I left a few comments.

**Comment by apc999** [comment]
Jenkis, test this please

**Comment by csliu** [comment]
addressed review comments.

File: common/src/main/resources/tachyon-default.properties [hunk 4]
```diff
@@ -54,6 +54,7 @@ tachyon.integration.worker.resource.mem=1024MB
 
 # Security properties
 tachyon.security.authentication.type=NOSASL
+tachyon.security.authentication.socket.timeout.second=30
```
Comment (apc999) [inline common/src/main/resources/tachyon-default.properties]: NIT: `second`-> `ms` and multiple the value by 1000?

**Comment by csliu** [comment]
fix another review and pass local full test.

**Comment by csliu** [comment]
@gpang updated the email setting, let me know if there still any issue.

**Comment by apc999** [comment]
+1 LGTM

**Comment by gpang** [comment]
ok to test

**Comment by csliu** [comment]
there seems still email issue. I will close this and redo it in another pr.

**Pull request closed without merging**